kale/embed/factorization.py

```diff
@@ -310,37 +312,43 @@ def _get_kernel(self, x, y=None):
             raise ValueError("Pre-computed kernel not supported")
         return pairwise_kernels(x, y, metric=self.kernel, filter_params=True, **params)
 
-    def fit(self, x, y=None, covariates=None):
+    def fit(self, x, y=None, groups=None):
         """
         Args:
             x : array-like. Input data, shape (n_samples, n_features)
-            y : array-like. Labels, shape (nl_samples,)
-            covariates : array-like. Domain co-variates, shape (n_samples, n_co-variates)
+            y : array-like. Labels, shape (n_labeled_samples,)
+            groups : array-like. Domain labels/grouping factors, shape (n_samples, )
 
         Note:
             Unsupervised MIDA is performed if y is None.
             Semi-supervised MIDA is performed is y is not None.
         """
-        if self.augmentation and isinstance(covariates, np.ndarray):
-            x = np.concatenate((x, covariates), axis=1)
+
+        if self.augmentation and isinstance(groups, np.ndarray):
+            x = np.concatenate((x, groups), axis=1)
+
+        groups = self.enc.fit_transform(groups.reshape(-1, 1)).toarray()
 
         # Kernel matrix
         kernel_x = self._get_kernel(x)
         kernel_x[np.isnan(kernel_x)] = 0
 
+        if self.n_components is None:
+            self.n_components = kernel_x.shape[0]
```

Comment:
recommend to add a warning in case if the kernel matrix's size is too large to do eigendecomposition. also, try to avoid mutating the defined attributes from `__init__` as it will causes an issue with `set_params(...)`. if `n_components` is too large, don't change it, rather just make a warning and then compute the eigenvalues with supported dimension size.

Reply:
updated in _fit()